network.go

```diff
@@ -1089,6 +1081,66 @@ func networkInfoListFromNetworkScan(handle *netlink.Handle) ([]NetworkInfo, erro
 	return netInfoList, nil
 }
 
+func createEndpointsFromScan(networkNSPath string) ([]Endpoint, error) {
+	var endpoints []Endpoint
+
+	netnsHandle, err := netns.GetFromPath(networkNSPath)
+	if err != nil {
+		return []Endpoint{}, err
+	}
+	defer netnsHandle.Close()
+
+	netlinkHandle, err := netlink.NewHandleAt(netnsHandle)
+	if err != nil {
+		return []Endpoint{}, err
+	}
+	defer netlinkHandle.Delete()
+
+	netInfoList, err := networkInfoListFromNetworkScan(netlinkHandle)
+	if err != nil {
+		return []Endpoint{}, err
+	}
+
+	uniqueID := uuid.Generate().String()
+
+	idx := 0
+	for _, netInfo := range netInfoList {
+		var endpoint Endpoint
+
+		// Skip any loopback interface.
+		if (netInfo.Iface.Flags & net.FlagLoopback) != 0 {
+			continue
+		}
+
+		if err := doNetNS(networkNSPath, func(_ ns.NetNS) error {
```

Comment:
A thought here. Given that netlink is fully namespace aware though the handle. If is pass the ns handle to the network function do we need this?

Reply:
Nope we cannot since Physical devices needs to be analyzed with ethtool and this has to happen from the netns.